iib/web/models.py

```diff
@@ -1075,6 +1095,170 @@ def get_mutable_keys(self):
+
+        :param dict kwargs: the JSON payload of the request.
+        :param Batch batch: the batch to specify with the request.
+        """
+        request_kwargs = deepcopy(kwargs)
+
+        deprecation_list = request_kwargs.pop('deprecation_list', [])
+        if not isinstance(deprecation_list, list) or any(
+            not item or not isinstance(item, str) for item in deprecation_list
+        ):
+            raise ValidationError(
+                '"deprecation_list" should be an empty array or an array of strings.'
+            )
+
+        request_kwargs['deprecation_list'] = [
+            Image.get_or_create(pull_specification=item) for item in deprecation_list
+        ]
+
+        source_from_index = request_kwargs.pop('source_from_index', None)
+        if not isinstance(source_from_index, str):
+            raise ValidationError('"source_from_index" must be a string')
+        request_kwargs['source_from_index'] = Image.get_or_create(
+            pull_specification=source_from_index
+        )
+
+        target_index = request_kwargs.pop('target_index', None)
+        if target_index:
+            if not isinstance(target_index, str):
+                raise ValidationError('"target_index" must be a string')
+            request_kwargs['target_index'] = Image.get_or_create(pull_specification=target_index)
+
+        # Verify that `overwrite_target_index` is the correct type
+        overwrite = request_kwargs.pop('overwrite_target_index', False)
+        if not isinstance(overwrite, bool):
+            raise ValidationError('The "overwrite_target_index" parameter must be a boolean')
+
+        # Verify that `overwrite_from_index_token` is the correct type
+        overwrite_token = request_kwargs.pop('overwrite_target_index_token', None)
+        if overwrite_token:
+            if not isinstance(overwrite_token, str):
+                raise ValidationError(
+                    'The "overwrite_target_index_token" parameter must be a string'
+                )
+            if overwrite_token and not overwrite:
+                raise ValidationError(
+                    'The "overwrite_target_index" parameter is required when'
+                    ' the "overwrite_target_index_token" parameter is used'
+                )
+
+        # Verify the user is authorized to use overwrite_target_index
+        # current_user.is_authenticated is only ever False when auth is disabled
+        if current_user.is_authenticated:
+            privileged_users = current_app.config['IIB_PRIVILEGED_USERNAMES']
+            if overwrite and not overwrite_token and current_user.username not in privileged_users:
+                raise Forbidden(
+                    'You must be a privileged user to set "overwrite_target_index" without'
+                    ' setting "overwrite_target_index_token"'
+                )
+
+        # Validate binary_image is correctly provided
+        binary_image = request_kwargs.pop('binary_image')
```

Comment:
To avoid a `KeyError`:
```suggestion
        binary_image = request_kwargs.pop('binary_image', None)
```